frontend/src/metabase/sharing/components/SharingSidebar.jsx

```diff
@@ -117,52 +112,45 @@ const getEditingPulseWithDefaults = (state, props) => {
 const mapStateToProps = (state, props) => ({
   pulseId: getPulseId(state, props),
   pulse: getEditingPulseWithDefaults(state, props),
-  cardPreviews: getPulseCardPreviews(state, props),
   formInput: getPulseFormInput(state, props),
   user: getUser(state),
   initialCollectionId: Collections.selectors.getInitialCollectionId(
     state,
     props,
   ),
-  pulseList: getPulseList(state, props),
 });
 
 const mapDispatchToProps = {
   setEditingPulse,
   updateEditingPulse,
   saveEditingPulse,
   fetchPulseFormInput,
-  fetchPulseCardPreview,
   setPulseArchived: Pulses.actions.setArchived,
   testPulse,
-  onChangeLocation: push,
-  goBack,
-  fetchPulsesByDashboardId,
 };
 
-@User.loadList()
+@Pulses.loadList({
+  query: (state, { dashboard }) => ({ dashboard_id: dashboard.id }),
+})
+@User.loadList({ loadingAndErrorWrapper: false })
 @connect(
   mapStateToProps,
   mapDispatchToProps,
 )
 class SharingSidebar extends React.Component {
   state = {
-    editingMode: undefined,
+    editingMode: "list-pulses",
     // use this to know where to go "back" to
-    returnMode: undefined,
+    returnMode: [],
```

Comment:
Why change this to a stack in this PR? Are there now >2 levels of modes?

Reply:
There always have been, we just didn't catch it yesterday since it's less common.

The specific path is `list-pulses` → `new-pulse` → `add-edit-slack` or `add-edit-email`.